velox/connectors/clp/search_lib/ir/ClpIrVectorLoader.cpp

```diff
@@ -33,12 +33,20 @@ void ClpIrVectorLoader::loadInternal(
     auto& logEvent = filteredLogEvents_->at(vectorIndex);
     // TODO: also need to support auto-generated keys
     auto userGenNodeIdValueMap = logEvent->get_user_gen_node_id_value_pairs();
-    auto const value_it{userGenNodeIdValueMap.find(nodeId_)};
-    if (userGenNodeIdValueMap.end() == value_it ||
-        false == value_it->second.has_value()) {
+    auto valueIt = userGenNodeIdValueMap.end();
+    ::clp::ffi::SchemaTree::Node::id_t nodeId{};
+    for (size_t i{0}; i < nodeIds_.size(); ++i) {
+      valueIt = userGenNodeIdValueMap.find(nodeIds_[i]);
+      if (valueIt != userGenNodeIdValueMap.end()) {
+        nodeId = nodeIds_[i];
+        break;
+      }
+    }
```

Comment:
```suggestion
    for (auto const candidateNodeId : nodeIds_) {
      valueIt = userGenNodeIdValueMap.find(candidateNodeId);
      if (valueIt != userGenNodeIdValueMap.end()) {
        nodeId = candidateNodeId;
        break;
      }
    }
```
Sort of up to taste, but I think this is a bit more readable.